interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala

```diff
@@ -167,25 +170,36 @@ class SpelExpressionSuggester(expressionConfig: ExpressionDefinition[_], typeDef
 }
 
 private class NuSpelNodeParser(typer: Typer) extends LazyLogging {
-  private val parser = new org.springframework.expression.spel.standard.SpelExpressionParser()
-
-  def parse(input: String, variables: Map[String, TypingResult]): Try[NuSpelNode] = {
-    val rawSpelExpression = Try(parser.parseRaw(input))
-    rawSpelExpression.map { parsedSpelExpression =>
-      val collectedTypingResult = typer.doTypeExpression(parsedSpelExpression, ValidationContext(variables))._2
-      val nuAst = new NuSpelNode(parsedSpelExpression.getAST, None, collectedTypingResult)
-      nuAst
+  private val parser = new NuTemplateAwareExpressionParser(new SpelParserConfiguration)
+
+  def parse(input: String, language: String, position: Int, variables: Map[String, TypingResult]): Try[Option[(NuSpelNode, Int)]] = {
+    val rawExpression = if (language == Expression.Language.Spel) {
+      Try(parser.parseExpression(input, null))
+    } else if (language == Expression.Language.SpelTemplate) {
+      Try(parser.parseExpression(input, new TemplateParserContext()))
+    } else {
+      Failure(new IllegalArgumentException(s"Language $language is not supported"))
+    }
```

Comment:
That looks like a perfect place for `match` and `case` :)

Reply:
Done! :)